src/main/webapp/app/entities/attachment/attachment.service.ts

```diff
@@ -50,6 +50,30 @@ export class AttachmentService {
         return this.http.delete<any>(`${this.resourceUrl}/${id}`, { observe: 'response' });
     }
 
+    downloadAttachment(downloadUrl: string) {
+        return new Observable(observer => {
```

Comment:
Creating a new Observable is always kind of ugly, because you then have to track inside it where it emits.

I would propose using switchMap, this also solves the shadowed name issue for response that appears twice in the observable:

```
    downloadAttachment(downloadUrl: string) {
        return this.http
            .get('api/files/attachments/access-token', { observe: 'response', responseType: 'text' })
            .pipe(
                switchMap(response => this.http.get(`${downloadUrl}?access_token=${response.body}`, { observe: 'response', responseType: 'blob' })),
                tap(response => {
                    const blob = new Blob([response.body!], { type: response.headers.get('content-type')! });
                    const url = window.URL.createObjectURL(blob);
                    const link = document.createElement('a');
                    link.setAttribute('href', url);
                    link.setAttribute('download', response.headers.get('filename')!);
                    document.body.appendChild(link); // Required for FF
                    link.click();
                    window.URL.revokeObjectURL(url);
                }),
            )
            .subscribe();
    }
```

Reply:
thanks for the input. i replaced it